internal/master/service.go

```diff
@@ -442,12 +450,44 @@ func (ds *distributedService) ListNodes(ctx context.Context, _ *empty.Empty) (*s
 }
 
 func (ds *distributedService) Close() error {
+	ds.lg.Info("Closing the master service")
 	// Do not invoke DKVService::Close here since `raftRepl` already
 	// closes the underlying storage connection.
+	// TODO - fix the above as ideally it should call DKVService::Close if there are more aspects to DKVService to close other than storage
 	ds.raftRepl.Stop()
+
+	ds.isClosed = true
 	return nil
 }
 
+func (ds *distributedService) GetStatus(context context.Context, request *emptypb.Empty) (*serverpb.RegionInfo, error) {
+	regionInfo := ds.DKVService.(*standaloneService).regionInfo
+
+	if ds.isClosed {
+		regionInfo.Status = serverpb.RegionStatus_INACTIVE
+	} else {
+		// Currently there is no way for this instance of DKVServer to know correctly if its the leader for its vBucket
+		// Neither does it know if its the local dc follower, or is a follower with a lot of lag
+		// Even the member list api ListMembers() is just an in memory lookup rather than actually looking at the current raft member state
+		// If the current node is itself disconnected, it will provide stale / incorrect member list
+		// For now, we will return status based on listMembers() and identify master based on IP address alone
+		// As of now, there is no hard requirement to identify true leader via service discovery thus using listmembers() is fine
```

Comment:
comment is mostly correct except the part of relying on IP address. removed that. 

basically the info is still stale as we're relying on in memory listNodes() function rather than check the raft quorum at the time. current use case its fine, but comment would still be required to call it out.

Reply:
ListNodes now returns actual/available info based on the node where this query happens. Leader info is always consistent.